pkg/agent/stream/eventlistener.go

```diff
@@ -97,19 +97,16 @@ func (em *EventListener) start() (done bool, err error) {
 
 // handleEvent fetches the api server ResourceClient based on the event self link, and then tries to save it to the cache.
 func (em *EventListener) handleEvent(event *proto.Event) error {
-	em.logger.
-		WithField("sequenceID", event.Metadata.SequenceID).
-		WithField("action", proto.Event_Type_name[int32(event.Type)]).
-		WithField("kind", event.Payload.Kind).
-		WithField("name", event.Payload.Name).
-		Debug("processing received watch event")
+	ctx := handler.NewEventContext(proto.Event_Type(event.Type), event.Metadata, event.Payload.Name, event.Payload.Kind)
```

Comment:
`event.Type` is already a `proto.Event_Type` so this conversion is unnecessary

Reply:
the casting from the previous log had me thinking it was an int32, you  right though. Fixed